.github/workflows/ruff.yml

```diff
@@ -0,0 +1,12 @@
+name: Ruff
+on: [ push, pull_request ]
```

Comment:
Right the check is running twice. Let's make it run:
* before integration (on pull request)
* after integration (on push, but only to master branch)